Cannot find module 'ts-node' (yarn)

<!-- Please don't delete this template otherwise your issue will be closed immediately -->
<!-- Before creating an issue please make sure you are using the latest version of webpack. -->

### Bug report

<!-- Please ask questions on discussions, StackOverflow or the webpack Gitter. -->
<!-- https://github.com/webpack/webpack/discussions -->
<!-- https://stackoverflow.com/questions/ask?tags=webpack -->
<!-- https://gitter.im/webpack/webpack -->
<!-- Issues which contain questions or support requests will be closed. -->

### Actual Behavior

Attempts to build using this loader result in the following:

```bash
! Module build failed: Error: Cannot find module 'ts-node'
ERR! Require stack:
ERR! - /home/circleci/project/node_modules/cosmiconfig-typescript-loader/dist/cjs/index.js
```

### Expected Behavior

https://github.com/Codex-/cosmiconfig-typescript-loader/blob/main/package.json#LL39C6-L39C13

cosmiconfig-typescript-loader has a peerDep of ts-node. This module has cosmiconfig-typescript-loader as a dependency - therefore it should specify ts-node as a dependency as well.

### How Do We Reproduce?

https://github.com/ntucker/anansi/pull/1941

https://app.circleci.com/pipelines/github/ntucker/anansi/9024/workflows/561c1fc3-70c1-403c-a033-8175c9788786/jobs/47985

### Please paste the results of `npx webpack info` here, and mention other relevant information

```bash
  System:
    OS: Linux 5.15 Ubuntu 20.04 LTS (Focal Fossa)
    CPU: (20) x64 AMD Ryzen 9 7950X 16-Core Processor
    Memory: 27.62 GB / 31.21 GB
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 3.5.0 - ~/.yarn/bin/yarn
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
  Monorepos:
    Yarn Workspaces: 3.5.0
  Packages:
    babel-loader: 9.1.2 => 9.1.2 
    webpack: 5.77.0 => 5.77.0 
    webpack-cli: 5.0.1 => 5.0.1 
    webpack-dev-server: 4.13.2 => 4.13.2 
```

I can't upgrade to `7.2.1` neither due to this error, found it in our webpack build. Also, we never use `ts-node` and I can't find `ts-node` installed by any packages in our `yarn.lock` so obviously we would bump into this `Error: Cannot find module 'ts-node'` problem.

As @ntucker has mentioned, 

> * [you use it right here](https://github.com/webpack-contrib/postcss-loader/blob/master/src/utils.js#L56)

if this `require` fails, why isn't it silently caught by the `try ... catch ... ` but bubble up all the way and cause webpack to fail?

--- a/package.json
+++ b/package.json
@@ -39,7 +39,17 @@
   ],
   "peerDependencies": {
     "postcss": "^7.0.0 || ^8.0.1",
-    "webpack": "^5.0.0"
+    "webpack": "^5.0.0",
+    "ts-node": ">=10",
+    "typescript": ">=4"
+  },
+  "peerDependenciesMeta": {
+    "ts-node": {
+      "optional": true
+    },
+    "typescript": {
+      "optional": true
+    }
   },
   "dependencies": {
     "cosmiconfig": "^8.1.3",

